src/Utils/OccRunner.php

```diff
@@ -32,16 +34,71 @@ class OccRunner {
 	protected $locator;
 
 	/**
+	 * @var bool
+	 */
+	protected $canUseProcess;
+
+	/**
 	 *
 	 * @param Locator $locator
 	 */
 	public function __construct(Locator $locator){
 		$this->locator = $locator;
+
+		$disabled = explode(',', ini_get('disable_functions'));
+		$this->canUseProcess = function_exists('proc_open') && !in_array('proc_open', $disabled);
+		$this->canUseProcess = false;
+	}
+
+	public function run($command, $args = [], $extra = ''){
+		if ($this->canUseProcess){
+			$cmdLine = trim($command . ' ' . $extra);
+			foreach ($args as $optionTitle => $optionValue){
+				if (strpos($optionTitle, '--') === 0){
+					$line = trim("$optionTitle $optionValue");
+				} else {
+					$line = $optionValue;
+				}
+				$escapedLine = ProcessUtils::escapeArgument($line);
+				$cmdLine .= " $escapedLine";
+			}
+			return $this->runAsProcess($cmdLine);
+		} else {
+			return $this->runAsRequest($command, $args);
+		}
+	}
+
+	public function runJson($command, $args = []){
+		$plain = $this->run($command, $args, '--output=json');
+		$decoded = json_decode($plain, true);
+		if (!is_array($decoded)){
+			throw new \UnexpectedValueException('Could not parse a response for ' . $command . '. Please check if the current shell user can run occ command. Raw output: ' . PHP_EOL . $plain);
+		}
+		return $decoded;
+	}
+
+	private function runAsRequest($command, $args){
+		$client = new Client();
+		$request = $client->createRequest(
+			'POST',
+			// TODO Get endpoint somehow :-/
+			'http://localhost/~deo/oc-tmp/index.php/occ/' . $command,
```

Comment:
well - we are only in web executor mode is the updater is executed from the browser.
Within the browser we know the address - right?